Assets/Scripts/Controllers/CharController.cs

```diff
@@ -198,7 +197,13 @@ protected bool MeleeAttack(LayerMask mask)
             {
                 //target.GetComponent<CharController>().TakeDamage(actualMeleeDamage);
                 //target.GetComponent<Rigidbody>().detectCollisions = false;
-                target.GetComponentInParent<CharController>().TakeDamage(actualMeleeDamage);
+                if (target.GetComponentInParent<EnemyController>())
```

Comment:
This code block should be reverted. It does the same thing as the one-liner it replaces